src/mqtt_client.rs

```diff
@@ -12,14 +12,21 @@ use embedded_nal::{nb, IpAddr, Mode, SocketAddr};
 use generic_array::{ArrayLength, GenericArray};
 use heapless::String;
 
+pub trait Clock {
+    /// Returns the current time, in seconds.
+    fn now(&self) -> u16;
+}
```

Comment:
Please use [`embedded-time`](https://crates.io/crates/embedded-time) instead

Reply:
That's a very sensible choice; I just didn't know whether there was a clear "standard" library for clocks in embedded scenarios yet, so went with the most minimal solution (a u16 seconds clock as required by MQTT). I'll make it depend on `embedded_time` instead.